src/chapters/more-on-functions/recursion.rst

```diff
@@ -0,0 +1,88 @@
+Recursion
+==========
+
+Quick Review
+-------------
+
+In the previous chapter, we learned how to define a function and its
+parameters.
+
+.. admonition:: Example
+
+   .. sourcecode:: js
+
+      function addTwoToNumber(num){
+         return num += 2;
+      }
+
+      console.log(addTwoToNumber(12));
+
+   **Output**
+
+   .. sourcecode:: js
+
+      14
+
+When called, the parameter ``num`` is passed an argument, which in this case is
+the number ``12``. The function executes and returns the value ``14``, which
+the ``console.log`` statement prints.
+
+Functions Can Call Other Functions
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Functions should only accomplish one (preferably simple) task. To solve more
+complicated tasks, one small function must call other functions.
+
+.. admonition:: Example
+
+   .. sourcecode:: js
+
+      function addTwoToNumber(num){
+         return num += 2;
+      }
+
+      function addFiveToNumber(value){
+         let result = addTwoToNumber(value) + 3;
+         return result;
+      }
+
+      console.log(addFiveToNumber(12))
+
+   **Output**
+
+   .. sourcecode:: js
+
+      17
+
+Of course, there is no need to write a function to add 5 to a value, but the
+example demonstrates calling a function from within another function.
+
+What is 'Recursion'?
+---------------------
+
+.. index:: ! recursion
+
+**Recursion** is the process of solving a larger problem by breaking it into
```

Comment:
I recognize that a function calling itself is one of the key ideas in recursion.  I felt, however, that a small lead-in from "identical pieces" to "calling itself" would help.  About 3 sentences after the "pieces" idea, I bring in the "calling itself" part.

We can reformulate the intervening text, but I like the idea of tying recursion to something the students are familiar with, namely loops.  Since a loop performs a single task multiple times, I think it provides a good mental model for what recursion accomplishes.

Reply:
d
ed
xed
ixed
Fixed